frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -576,7 +617,37 @@ function ReaderHighlight:_resetHoldTimer(clear)
     end
 end
 
+function ReaderHighlight:onTogglePanelZoomSetting(arg, ges)
+    if not self.document.info.has_pages then return end
+    self.panel_zoom_enabled = not self.panel_zoom_enabled
+    self:onSaveSettings()
+end
+
+function ReaderHighlight:onPanelZoom(arg, ges)
+    self:clear()
+    local hold_pos = self.view:screenToPageTransform(ges.pos)
+    if not hold_pos then return false end -- outside page boundary
+    local rect = self.ui.document:getPanelFromPage(hold_pos.page, hold_pos)
+    if not rect then return false end -- panel not found, return
+    local image = self.ui.document:getPagePart(hold_pos.page, rect, 0)
+
+    if image then
+        local ImageViewer = require("ui/widget/imageviewer")
+        local imgviewer = ImageViewer:new{
+            image = image,
+            with_title_bar = false,
+            fullscreen = true,
+        }
+        UIManager:show(imgviewer)
+    end
+    return true
+end
+
 function ReaderHighlight:onHold(arg, ges)
+    if self.panel_zoom_enabled and self.document.info.has_pages then
```

Comment:
(nitpick: logical order, just for readability: first self.document.info.has_pages , and then self.panel_zoom_enabled  that makes sense only then.)